StreemNow/ViewController.swift

```diff
@@ -165,6 +193,36 @@ class ViewController: UIViewControllerSupport {
         }
         
     }
+    
+    func createInvite() {
+        if let currentUser = StreemInitializer.shared.currentUser {
+            var name = "StreemNow Invitee"
+            let currentUserName = currentUser.name
+            let initials = currentUserName.split(separator: " ")
+                .compactMap { $0.prefix(1).uppercased() }
+                .reduce("") { "\($0)\($1)" }
+            name += " (\(initials))"
+
+            activityIndicator?.startAnimating()
+            
+            Streem.sharedInstance.createInvitation(forUser: name, referenceId: nil, type: .link) { [weak self] invitation, error in
+                DispatchQueue.main.async { [weak self] in
+                    guard let self = self else { return }
+                    self.showActivityIndicator(false)
+                    
+                    if let invitation = invitation {
+                        DispatchQueue.main.async { [weak self] in
+                            guard let self = self else { return }
+                            self.invitationTextField.text = invitation.code.formattedAsInvitationCode()
+                            self.activityIndicator?.stopAnimating()
+                        }
+                    } else {
+                        self.presentAlert(message: "Error creating invitation - \(error.debugDescription)")
+                    }
+                }
```

Comment:
Why introduce the new `activityIndicator`? Can't you instead call `self.showActivityIndicator(true)` before calling `createInvitation()`, and then move `self.showActivityIndicator(false)` into both the spot where you now have `self.activityIndicator?.stopAnimating()` and also into the `else` clause?

Reply:
Nice! I didn't realize we already had one available.

Reply:
@dgoldman-pdx Done